server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1256,13 +1256,24 @@ private static boolean isFiltered(Property[] properties) {
     public static class RegexValidator implements Writeable, Validator<String> {
         private Pattern pattern;
 
+        private boolean isMatching = true;
+
         /**
          * @param regex A regular expression containing the only valid input for this setting.
          */
         public RegexValidator(String regex) {
             this.pattern = Pattern.compile(regex);
         }
 
+        /**
+         * @param regex A regular expression containing the only valid input for this setting.
+         * @param isMatching this is a boolean switching between "fail, if it matches" or "fail, if it doesn't match".
+         */
+        public RegexValidator(String regex, boolean isMatching) {
+            this.pattern = Pattern.compile(regex);
+            this.isMatching = isMatching;
+        }
+
         public RegexValidator(StreamInput in) throws IOException {
             this.pattern = Pattern.compile(in.readString());
         }
```

Comment:
You need to add the boolean to this constructor so it can be read when sent over transport.  Also you need to add it (in the same order) to the `writeTo()` method.